grp/classic.gd

```diff
@@ -227,19 +236,59 @@ DeclareConstructor( "GeneralOrthogonalGroupCons",
 ##  <#/GAPDoc>
 ##
 BindGlobal( "GeneralOrthogonalGroup", function ( arg )
```

Comment:
I think the code below could be simplified quite a bit (roughly halving the code?) by doing something like this:
```suggestion
BindGlobal( "GeneralOrthogonalGroup", function ( arg )
  local filter;
  if IsFilter(arg[1]) then
    filter := Remove(arg, 1);
  else
    filter := IsMatrixGroup;
  fi;
```
And then using `filter` in the obvious way.

One could also do something like this (in addition):
```gap
  if IS_FORM_OR_GROUP_WITH_FORM(Last(arg)) then
     form := Remove(arg);
     # now handle the cases with form -- as I state above, I would assume at this
     # point either `arg` is empty, or else contains a ring or the size of a finite field
     ...
  fi;

Reply:
Yes.
(One could have done this already before the changes proposed here, but it was less striking.)